kubernetes/kubernetes.go

```diff
@@ -219,12 +218,13 @@ func performConnectionChecks(ka *KubeAgentConfig) error {
 		ProxyAuth:     ka.OutboundProxyAuth,
 		ProxyInsecure: ka.OutboundProxyInsecure,
 		Timeout:       time.Duration(ka.HTTPSTimeout) * time.Second,
+		Region:        ka.UploadRegion,
 	})
 	if err != nil {
 		return errors.New("error creating Cloudability Metric client in connectivity test")
 	}
 
-	metricSampleURL := apiEndpoint + "/metricsample"
+	metricSampleURL := client.GetUploadURL(ka.UploadRegion) + "/metricsample"
```

Comment:
Nitpick, could we move `metricsample` within the GetUploadURL method...

Reply:
The upload URL is logged without this /metricsample in an error log. Not sure if we care that this would change but I figured just leave as is to reduce change radius

Reply:
Is there any concern to log out the `metricsample` as well.
The idea for this comment is to have unified func to generate the URL to minimize the risk that we test URL_A here and actual getURL is using URL_B because it is generated in different way

Reply:
No concern with logging /metricsample. My concern is with extensiblity. It looks like this url was originally committed to support several endpoiints but ended up only needing /metricsample. For example there is a SendMeasurement function that hits /metrics https://github.com/cloudability/metrics-agent/blob/2c8c5421b0ceae0e556398d0560af641448d8ff0/client/client.go#L157

It seems unused expect by test files and is from the original commit. I can add the /metricsample to the base url since I doubt we have any need for other endpoints in the MA

Reply:
I feel changing this base url is not as simple as a 1 line change as some testing structure is based on it. So I don't really see the value. If it is a blocker to merge then I will rewrite the test files to support the base url having `/metricsample` if needed.

Reply:
I do get your point though that this is still two separate url generations. Would a method like getMetricsSampleURL() that returns a string with the entire url be better in your opinion. So that we can do something like
`metricSampleURL := getMetricsSampleURL(client.GetUploadURL(ka.UploadRegion))` in both places so we know generation is the same as we use the same method.

Reply:
For test, could we just update baseURL in the code to match?
https://github.com/cloudability/metrics-agent/blob/78faf44041d249581fec5d4361b8d7dd419a4f6c/client/client_test.go#L111
And current test seems doesn't cover the region specific logic and we only care about the URI suffix is `/metrcisample` , which seems a gap. 

I would in favor of making our code cleaner rather than worry about the extensibility to a non-valid use case (SendMeasurement).